apiserver/facades/agent/meterstatus/meterstatus_test.go

```diff
@@ -91,5 +94,113 @@ func (s *meterStatusSuite) TestGetMeterStatus(c *gc.C) {
-	meterstatustesting.TestWatchMeterStatus(c, s.status, s.unit, s.State, s.resources)
+	status, ctrl := s.setupMeterStatusAPI(c, func(st *mocks.MockMeterStatusState, res *mocks.MockResources, auth *mocks.MockAuthorizer) {
+		aExp := auth.EXPECT()
+		sExp := st.EXPECT()
+		rExp := res.EXPECT()
+
+		tag := s.unit.UnitTag()
+		aExp.GetAuthTag().Return(tag)
+
+		aExp.AuthOwner(tag).Return(true)
+		sExp.Unit(tag.Id()).Return(s.unit, nil)
+		rExp.Register(gomock.Any()).Return("1")
+
+		aExp.AuthOwner(names.NewUnitTag("foo/42")).Return(false)
+	})
+	defer ctrl.Finish()
+
+	args := params.Entities{Entities: []params.Entity{
+		{Tag: s.unit.UnitTag().String()},
+		{Tag: "unit-foo-42"},
+	}}
+	result, err := status.WatchMeterStatus(args)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, gc.DeepEquals, params.NotifyWatchResults{
+		Results: []params.NotifyWatchResult{
+			{NotifyWatcherId: "1"},
+			{Error: apiservertesting.ErrUnauthorized},
+		},
+	})
+}
+
+func (s *meterStatusSuite) TestWatchMeterStatusWithStateChange(c *gc.C) {
+	status, ctrl := s.setupMeterStatusAPI(c, func(st *mocks.MockMeterStatusState, res *mocks.MockResources, auth *mocks.MockAuthorizer) {
+		aExp := auth.EXPECT()
+		sExp := st.EXPECT()
+		rExp := res.EXPECT()
+
+		tag := s.unit.UnitTag()
+		aExp.GetAuthTag().Return(tag)
+
+		aExp.AuthOwner(tag).Return(true)
+		sExp.Unit(tag.Id()).Return(s.unit, nil)
+		rExp.Register(gomock.Any()).Return("1")
+	})
+	defer ctrl.Finish()
+
+	args := params.Entities{Entities: []params.Entity{
+		{Tag: s.unit.UnitTag().String()},
+	}}
+	result, err := status.WatchMeterStatus(args)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, gc.DeepEquals, params.NotifyWatchResults{
+		Results: []params.NotifyWatchResult{
+			{NotifyWatcherId: "1"},
+		},
+	})
+}
+
+func (s *meterStatusSuite) TestWatchMeterStatusWithApplicationTag(c *gc.C) {
+	app, err := s.unit.Application()
+	c.Assert(err, jc.ErrorIsNil)
+
+	units, err := app.AllUnits()
+	c.Assert(err, jc.ErrorIsNil)
+
+	unit := units[0]
+
+	status, ctrl := s.setupMeterStatusAPI(c, func(st *mocks.MockMeterStatusState, res *mocks.MockResources, auth *mocks.MockAuthorizer) {
+		aExp := auth.EXPECT()
+		sExp := st.EXPECT()
+		rExp := res.EXPECT()
+
+		tag := names.ApplicationTag{
+			Name: "mysql",
+		}
+		aExp.GetAuthTag().Return(tag)
+
+		sExp.Application(tag.Name).Return(app, nil)
+		sExp.Unit(unit.Tag().Id()).Return(unit, nil)
+		rExp.Register(gomock.Any()).Return("1")
+	})
+	defer ctrl.Finish()
+
+	args := params.Entities{Entities: []params.Entity{
+		{Tag: unit.UnitTag().String()},
+	}}
+	result, err := status.WatchMeterStatus(args)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, gc.DeepEquals, params.NotifyWatchResults{
+		Results: []params.NotifyWatchResult{
+			{NotifyWatcherId: "1"},
+		},
+	})
+}
+
+func (s *meterStatusSuite) setupMeterStatusAPI(c *gc.C, fn func(*mocks.MockMeterStatusState, *mocks.MockResources, *mocks.MockAuthorizer)) (*meterstatus.MeterStatusAPI, *gomock.Controller) {
```

Comment:
This is subjective, but I would make a struct with the mocks as members. 

Then you can shorten up the signature on all the calls, plus you don't have to change them all if say another mock or testing artefact was added.